core/templates/pages/exploration-editor-page/services/graph-data.service.ts

```diff
@@ -29,16 +29,16 @@ angular.module('oppia').factory('GraphDataService', [
       ComputeGraphService, ExplorationInitStateNameService,
       ExplorationStatesService) {
     var _graphData = null;
-
-    // Returns an object which can be treated as the input to a visualization
-    // for a directed graph. The returned object has the following keys:
-    //   - nodes: an object whose keys are node ids (equal to node names) and
-    //       whose values are node names
-    //   - links: a list of objects. Each object represents a directed link
-    //       between two nodes, and has keys 'source' and 'target', the values
-    //       of which are the names of the corresponding nodes.
-    //   - initStateName: the name of the initial state.
-    //   - finalStateName: the name of the final state.
+    /**
+     * @return {Object} graphData - Direct graph visuapization input.
+     * @return {Object} graphData.nodes - A nodes object.
+     * @return {Object[]} graphData.links - List of direct links two of nodes.
+     * @return {string[]} graphData.finalStateIds - An array of state names.
+     * @return {string} graphData.initStateId - The name of initial state.
+     * @return {string} graphData.nodes.<stateName> - The name of the state.
+     * @return {string} graphData.links[i].source - The source node.
+     * @return {string} graphData.links[i].target - The target node.
+     */
```

Comment:
This looks incorrect. `_recomputeGraphData` does not actually return anything, right?

It should probably be before `getGraphData`. Also, I don't think you need to include the type info in the JSDocs per https://github.com/oppia/oppia/issues/8668#issue-566518484. Please take a look.

Reply:
Done